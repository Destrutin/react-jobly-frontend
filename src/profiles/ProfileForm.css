.Profile {
    max-width: 400px;
    margin: 0 auto;
    padding: 2rem;
    background-color: #e6e6fa;
    border: 1px solid #9370db;
    border-radius: 5px;
  }
  
  .profile-card {
    margin-bottom: 1rem;
  }
  
  .profile-input {
    margin-bottom: 1rem;
  }
  
  .profile-input label {
    display: block;
    margin-bottom: 0.5rem;
  }
  
  .profile-input input {
    width: 100%;
    padding: 0.5rem;
    border: 1px solid #ccc;
    border-radius: 5px;
  }
  
  .profile-submit {
    padding: 0.5rem 1rem;
    background-color: #007bff;
    color: #fff;
    border: none;
    border-radius: 5px;
    cursor: pointer;
  }
  
  .profile-submit:hover {
    background-color: #0056b3;
  }
  