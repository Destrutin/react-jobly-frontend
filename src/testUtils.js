import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  password: "omgsecret",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, hasAppliedToJob = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasAppliedToJob }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
