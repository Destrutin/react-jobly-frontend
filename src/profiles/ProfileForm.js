import React, { useContext, useState } from "react";
import Api from "../api/Api";
import UserContext from "../auth/UserContext";
import "./ProfileForm.css"

function ProfileForm() {
    const {currentUser, setCurrentUser} = useContext(UserContext);
    const [formData, setFormData] = useState({
        username: currentUser.username,
        password: '',
        firstName: currentUser.firstName,
        lastName: currentUser.lastName,
        email: currentUser.email,
    })
    const [updateSuccessful, setUpdateSuccessful] = useState(false);

    const handleChange = e => {
        const {name, value} = e.target;
        setFormData(data => ({...data, [name]:value}))
    }

    const handleSubmit = async(e) => {
        e.preventDefault();
        let profileData = {
            firstName: formData.firstName,
            lastName: formData.lastName,
            email: formData.email,
            password: formData.password
        };
        let username = formData.username;
        let updatedUser;

        try{
            updatedUser = await Api.saveChanges(username, profileData);
        } catch (err) {
            console.error(err);
        }

        setFormData(data => ({...data}));
        setCurrentUser(updatedUser);
        setUpdateSuccessful(true);
    }

    return (
        <div className="Profile">
            <h1>Profile</h1>
            <div className="profile-card">
                <div className="profile-input">
                    <label htmlFor="username">Username</label>
                    <input id="username" value={formData.username} disabled/>
                </div>

                <div className="profile-input">
                    <label htmlFor="firstName">First Name</label>
                    <input
                        id="firstName"
                        name="firstName"
                        value={formData.firstName}
                        onChange={handleChange}
                    />
                </div>

                <div className="profile-input">
                    <label htmlFor="lastName">Last Name</label>
                    <input
                        id="lastName"
                        name="lastName"
                        value={formData.lastName}
                        onChange={handleChange}
                    />
                </div>

                <div className="profile-input">
                    <label htmlFor="email">Email</label>
                    <input
                        id="email"
                        name="email"
                        value={formData.email}
                        onChange={handleChange}
                    />
                </div>
                {updateSuccessful ? alert("Updated successfully.") : null}

                <button className="profile-submit" onClick={handleSubmit}>
                  Save Changes  
                </button>
            </div>
        </div>
    )
}

export default ProfileForm;