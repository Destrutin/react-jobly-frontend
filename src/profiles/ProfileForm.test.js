import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import ProfileForm from './ProfileForm';
import {UserProvider} from '../testUtils';

test('renders profile form without crashing', () => {
    const {getByText, getByLabelText} = render(
        <UserProvider>
            <ProfileForm/>
        </UserProvider>
    )
    expect(getByText('Profile')).toBeInTheDocument();
    expect(getByLabelText('Username')).toBeInTheDocument();
    expect(getByLabelText('First Name')).toBeInTheDocument();
    expect(getByLabelText('Last Name')).toBeInTheDocument();
    expect(getByLabelText('Email')).toBeInTheDocument();
    expect(getByText('Save Changes')).toBeInTheDocument();
})

test('matches snapshot', () => {
    const {asFragment} = render(
        <UserProvider>
            <ProfileForm/>
        </UserProvider>
    )
    expect(asFragment()).toMatchSnapshot();
})

test('profile form submits with correct input', async () => {
    const {getByText, getByLabelText} = render(
        <UserProvider>
            <ProfileForm/>
        </UserProvider>
    )

    fireEvent.change(getByLabelText('First Name'), {target: {value: 'Fang'}});
    fireEvent.change(getByLabelText('Last Name'), {target: {value: 'Boi'}});
    fireEvent.change(getByLabelText('Email'), {target: {value: 'FangBoi@gmail.com'}});
    fireEvent.click(getByText('Save Changes'));
})