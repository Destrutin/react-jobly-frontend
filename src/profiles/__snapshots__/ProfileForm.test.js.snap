// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Profile"
  >
    <h1>
      Profile
    </h1>
    <div
      class="profile-card"
    >
      <div
        class="profile-input"
      >
        <label
          for="username"
        >
          Username
        </label>
        <input
          disabled=""
          id="username"
          value="testuser"
        />
      </div>
      <div
        class="profile-input"
      >
        <label
          for="firstName"
        >
          First Name
        </label>
        <input
          id="firstName"
          name="firstName"
          value=""
        />
      </div>
      <div
        class="profile-input"
      >
        <label
          for="lastName"
        >
          Last Name
        </label>
        <input
          id="lastName"
          name="lastName"
          value=""
        />
      </div>
      <div
        class="profile-input"
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          value="test@test.net"
        />
      </div>
      <button
        class="profile-submit"
      >
        Save Changes
      </button>
    </div>
  </div>
</DocumentFragment>
`;
